src/utils_request.js

```diff
@@ -0,0 +1,272 @@
+ *
+ * This function is useful for web scraping of websites that send the full HTML in the first response.
+ * Thanks to this function, the target web server has no simple way to find out the request
+ * hasn't been sent by a full web browser. Using a headless browser for such requests
+ * is an order of magnitude more resource-intensive than this function.
+ *
+ * Currently, the function sends requests the same way as Firefox web browser does.
+ * In the future, it might add support for other browsers too.
+ *
+ * Internally, the function uses `requestBetter()` function to perform the request.
+ * All `options` not recognized by this function are passed to it,
+ * so see it for more details.
+ *
+ * @param options.url
+ *  URL of the target endpoint. Supports both HTTP and HTTPS schemes.
+ * @param [options.method=GET]
+ *  HTTP method.
+ * @param [options.headers={}]
+ *  Additional HTTP headers to add. It's recommended not to use this option,
+ *  because it can ruin the signature of the web browser. TODO: Maybe let's remove this completely?
+ * @param [options.languageCode=en]
+ *  Two-letter ISO 639 language code.
+ * @param [options.countryCode=US]
+ *  Two-letter ISO 3166 country code.
+ * @param [options.isMobile]
+ *  If `true`, the function uses User-Agent of a mobile browser.
+ *
+ * @return {{ response, body }}
+ *  Returns an object with two properties: `response` is the instance of
+ *  Node's [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage) class,
+ *  `body` is a `String`, `Buffer` or `Object`, depending on the `encoding` and `json` options.
+ */
+export const requestLikeBrowser = async (options) => {
+    const defaultOptions = {
+        countryCode: 'US',
+        languageCode: 'en',
+        headers: {},
+        method: 'GET',
+        useMobileVersion: false,
+    };
+    const opts = Object.assign(defaultOptions, options);
+
+    const parsedUrl = url.parse(opts.url);
+
+    const browserHeaders = {
+        Host: parsedUrl.host,
+        'User-Agent': opts.useMobileVersion ? FIREFOX_MOBILE_USER_AGENT : FIREFOX_DESKTOP_USER_AGENT,
+        Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+        'Accept-Language': `${opts.languageCode}-${opts.countryCode},${opts.languageCode};q=0.5`,
+        'Accept-Encoding': 'gzip, deflate, br',
+        DNT: '1',
+        Connection: 'keep-alive',
+        'Upgrade-Insecure-Requests': '1',
+    };
+    opts.headers = Object.assign(browserHeaders, opts.headers);
+
+    const response = await requestBetter(opts);
+    const { type } = contentType.parse(response.headers['content-type']);
+
+    // Handle situations where the server explicitly states that
+    // it will not serve the resource as text/html by skipping.
+    if (response.statusCode === 406) {
+        throw new Error(`requestLikeBrowser: Resource ${options.url} is not available in HTML format. Skipping resource.`);
```

Comment:
`Skipping resource` should not be there.

Reply:
It should be in the `requestBetter`. However, I am not sure if `requestBetter` should use HTML specific things. I might check the request `Accept` header first, and if it is HTML, I could make this check.

Reply:
We'll need to do it somehow in `CheerioCrawler` so I guess having an option in `requestBetter` to abort certain `content-type` could be handy. This could then be used in `requestAsBrowser` for `text/html` specifically and `CheerioCrawler` would just use that.

Reply:
I think that I could do this in the `abortFunction` 💡